[Gradle Plugin] Support Configuration Cache

Current version doesn't support a Gradle [Configuration Cache](https://docs.gradle.org/current/userguide/configuration_cache.html).

Steps to reproduce a problem:
1. Download `hello-world` example https://github.com/playframework/play-samples/releases/download/nightly/play-java-hello-world-gradle-example.zip

2. Run any task with an enabled configuration cache. For example, `classes`
```
./gradlew classes --configuration-cache --stacktrace

* Exception is:
org.gradle.api.GradleException: Could not load the value of field `patterns` of `org.gradle.internal.serialize.codecs.core.DirectoryTreeSpec` bean found in field `mainSpec` of task `:processResources` of type `org.gradle.language.jvm.tasks.ProcessResources`.
        at org.gradle.internal.serialize.graph.BeanPropertyExtensionsKt.readPropertyValue(BeanPropertyExtensions.kt:68)
        at org.gradle.internal.serialize.graph.BeanPropertyExtensionsKt$readPropertyValue$1.invokeSuspend(BeanPropertyExtensions.kt)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlin.coroutines.SafeContinuation.resumeWith(SafeContinuationJvm.kt:41)
        at org.gradle.internal.serialize.graph.CombinatorsKt$reentrant$1$decodeLoop$$inlined$Continuation$1.resumeWith(Continuation.kt:78)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
        at kotlin.coroutines.ContinuationKt.startCoroutine(Continuation.kt:115)
        at org.gradle.internal.serialize.graph.CombinatorsKt$reentrant$1.decodeLoop(Combinators.kt:123)
        at org.gradle.internal.serialize.graph.CombinatorsKt$reentrant$1.decode(Combinators.kt:85)
        at org.gradle.internal.serialize.graph.codecs.BindingsBackedCodec.decode(BindingsBackedCodec.kt:70)
        at org.gradle.internal.serialize.graph.DefaultReadContext.read(Contexts.kt:279)
        at org.gradle.internal.serialize.graph.CombinatorsKt.readCollectionInto(Combinators.kt:189)
        at org.gradle.internal.serialize.codecs.stdlib.CollectionCodecsKt$collectionCodec$2.invokeSuspend(CollectionCodecs.kt:81)
        at org.gradle.internal.serialize.codecs.stdlib.CollectionCodecsKt$collectionCodec$2.invoke(CollectionCodecs.kt)
        at org.gradle.internal.serialize.codecs.stdlib.CollectionCodecsKt$collectionCodec$2.invoke(CollectionCodecs.kt)
        at org.gradle.internal.serialize.graph.CombinatorsKt$codec$1.decode(Combinators.kt:44)
        at org.gradle.internal.serialize.graph.codecs.BindingsBackedCodec.decode(BindingsBackedCodec.kt:70)
        at org.gradle.internal.serialize.graph.DefaultReadContext.read(Contexts.kt:279)
        at org.gradle.internal.serialize.graph.CodecKt.readNonNull(Codec.kt:156)
        at org.gradle.internal.serialize.codecs.core.FileTreeCodec.decode(FileTreeCodec.kt:98)
        at org.gradle.internal.serialize.graph.codecs.BindingsBackedCodec.decode(BindingsBackedCodec.kt:70)
        at org.gradle.internal.serialize.graph.DefaultReadContext.read(Contexts.kt:279)
        at org.gradle.internal.serialize.graph.CombinatorsKt.readList(Combinators.kt:174)
        at org.gradle.internal.serialize.codecs.core.FileCollectionCodec.decodeContents(FileCollectionCodec.kt:92)
        at org.gradle.internal.serialize.codecs.core.ConfigurableFileCollectionCodec.decode(ConfigurableFileCollectionCodec.kt:45)
        at org.gradle.internal.serialize.graph.codecs.BindingsBackedCodec.decode(BindingsBackedCodec.kt:70)
        at org.gradle.internal.serialize.graph.DefaultReadContext.read(Contexts.kt:279)
        at org.gradle.internal.serialize.codecs.core.DefaultCopySpecCodec.decode(DefaultCopySpecCodec.kt:71)
        at org.gradle.internal.serialize.graph.codecs.BindingsBackedCodec.decode(BindingsBackedCodec.kt:70)
        at org.gradle.internal.serialize.graph.DefaultReadContext.read(Contexts.kt:279)
        at org.gradle.internal.serialize.graph.BeanPropertyExtensionsKt.readPropertyValue(BeanPropertyExtensions.kt:61)
        at org.gradle.internal.serialize.beans.services.BeanPropertyReader.readFieldOf(BeanPropertyReader.kt:93)
        at org.gradle.internal.serialize.beans.services.BeanPropertyReader.readFieldsOf(BeanPropertyReader.kt:73)
        at org.gradle.internal.serialize.beans.services.BeanPropertyReader.readStateOf(BeanPropertyReader.kt:66)
        at org.gradle.internal.serialize.codecs.core.TaskNodeCodec$readTask$2.invokeSuspend(TaskNodeCodec.kt:134)
        at org.gradle.internal.serialize.codecs.core.TaskNodeCodec$readTask$2.invoke(TaskNodeCodec.kt)
        at org.gradle.internal.serialize.codecs.core.TaskNodeCodec$readTask$2.invoke(TaskNodeCodec.kt)
        at org.gradle.internal.serialize.codecs.core.TaskNodeCodecKt.withTaskOf(TaskNodeCodec.kt:236)
        at org.gradle.internal.serialize.codecs.core.TaskNodeCodecKt.access$withTaskOf(TaskNodeCodec.kt:1)
        at org.gradle.internal.serialize.codecs.core.TaskNodeCodec.readTask(TaskNodeCodec.kt:128)
        at org.gradle.internal.serialize.codecs.core.TaskNodeCodec.decode(TaskNodeCodec.kt:79)
        at org.gradle.internal.serialize.graph.codecs.BindingsBackedCodec.decode(BindingsBackedCodec.kt:70)
        at org.gradle.internal.serialize.graph.DefaultReadContext.read(Contexts.kt:279)
        at org.gradle.internal.serialize.graph.CodecKt.readNonNull(Codec.kt:156)
        at org.gradle.internal.serialize.codecs.core.WorkNodeCodec.readNode(WorkNodeCodec.kt:405)
        at org.gradle.internal.serialize.codecs.core.WorkNodeCodec.readGroupedNodes(WorkNodeCodec.kt:347)
        at org.gradle.internal.serialize.codecs.core.WorkNodeCodec.access$readGroupedNodes(WorkNodeCodec.kt:81)
        at org.gradle.internal.serialize.codecs.core.WorkNodeCodec$readNodes$1$1$1$1.invokeSuspend(WorkNodeCodec.kt:252)
        at org.gradle.internal.serialize.codecs.core.WorkNodeCodec$readNodes$1$1$1$1.invoke(WorkNodeCodec.kt)
        at org.gradle.internal.serialize.codecs.core.WorkNodeCodec$readNodes$1$1$1$1.invoke(WorkNodeCodec.kt)
        at org.gradle.internal.serialize.graph.CodecKt$readWith$1$1.invokeSuspend(Codec.kt:144)
        at org.gradle.internal.serialize.graph.CodecKt$readWith$1$1.invoke(Codec.kt)
        at org.gradle.internal.serialize.graph.CodecKt$readWith$1$1.invoke(Codec.kt)
        at org.gradle.internal.serialize.graph.RunningKt$runReadOperation$2.invokeSuspend(Running.kt:33)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlin.coroutines.ContinuationKt.startCoroutine(Continuation.kt:115)
        at org.gradle.internal.serialize.graph.RunningKt.runToCompletion(Running.kt:54)
        at org.gradle.internal.serialize.graph.RunningKt.runReadOperation(Running.kt:32)
        at org.gradle.internal.serialize.graph.CodecKt.readWith(Codec.kt:143)
        at org.gradle.internal.serialize.codecs.core.WorkNodeCodec$readNodes$1$1$1.invoke(WorkNodeCodec.kt:251)
        at org.gradle.internal.serialize.codecs.core.WorkNodeCodec$readNodes$1$1$1.invoke(WorkNodeCodec.kt:250)
        at org.gradle.internal.serialize.codecs.core.WorkNodeCodecKt$asBuildOperation$1.run(WorkNodeCodec.kt:634)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:67)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:60)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:167)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:60)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$QueueWorker.execute(DefaultBuildOperationExecutor.java:161)
        at org.gradle.internal.operations.DefaultBuildOperationQueue$WorkerRunnable.runOperation(DefaultBuildOperationQueue.java:272)
        at org.gradle.internal.operations.DefaultBuildOperationQueue$WorkerRunnable.doRunBatch(DefaultBuildOperationQueue.java:253)
        at org.gradle.internal.operations.DefaultBuildOperationQueue$WorkerRunnable.lambda$runBatch$1(DefaultBuildOperationQueue.java:226)
        at org.gradle.internal.work.DefaultWorkerLeaseService.runAsWorkerThread(DefaultWorkerLeaseService.java:125)
        at org.gradle.internal.operations.DefaultBuildOperationQueue$WorkerRunnable.runBatch(DefaultBuildOperationQueue.java:224)
        at org.gradle.internal.operations.DefaultBuildOperationQueue$WorkerRunnable.run(DefaultBuildOperationQueue.java:192)
        at org.gradle.internal.operations.DefaultBuildOperationQueue.waitForCompletion(DefaultBuildOperationQueue.java:105)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.executeInParallel(DefaultBuildOperationExecutor.java:106)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.runAllWithAccessToProjectState(DefaultBuildOperationExecutor.java:75)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.runAllWithAccessToProjectState(DefaultBuildOperationExecutor.java:70)
        at org.gradle.internal.serialize.codecs.core.WorkNodeCodec$runBuildOperations$1.invoke(WorkNodeCodec.kt:582)
        at org.gradle.internal.serialize.codecs.core.WorkNodeCodec$runBuildOperations$1.invoke(WorkNodeCodec.kt:581)
        at org.gradle.internal.serialize.codecs.core.WorkNodeCodec.handleBuildOperationExceptions(WorkNodeCodec.kt:281)
        at org.gradle.internal.serialize.codecs.core.WorkNodeCodec.runBuildOperations(WorkNodeCodec.kt:581)
        at org.gradle.internal.serialize.codecs.core.WorkNodeCodec.readNodes(WorkNodeCodec.kt:248)
        at org.gradle.internal.serialize.codecs.core.WorkNodeCodec.doRead(WorkNodeCodec.kt:131)
        at org.gradle.internal.serialize.codecs.core.WorkNodeCodec.readWork(WorkNodeCodec.kt:101)
        at org.gradle.internal.cc.impl.ConfigurationCacheState.readWorkGraph(ConfigurationCacheState.kt:521)
        at org.gradle.internal.cc.impl.ConfigurationCacheState.readBuildContent$configuration_cache(ConfigurationCacheState.kt:502)
        at org.gradle.internal.cc.impl.ConfigurationCacheState.readBuildState(ConfigurationCacheState.kt:353)
        at org.gradle.internal.cc.impl.ConfigurationCacheState.readBuildsInTree(ConfigurationCacheState.kt:315)
        at org.gradle.internal.cc.impl.ConfigurationCacheState.readRootBuild(ConfigurationCacheState.kt:288)
        at org.gradle.internal.cc.impl.ConfigurationCacheState.readRootBuildState(ConfigurationCacheState.kt:189)
        at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO$readRootBuildStateFrom$1.invokeSuspend(DefaultConfigurationCacheIO.kt:227)
        at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO$readRootBuildStateFrom$1.invoke(DefaultConfigurationCacheIO.kt)
        at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO$readRootBuildStateFrom$1.invoke(DefaultConfigurationCacheIO.kt)
        at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO$readConfigurationCacheStateWithSpecialDecoders$1.invokeSuspend(DefaultConfigurationCacheIO.kt:351)
        at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO$readConfigurationCacheStateWithSpecialDecoders$1.invoke(DefaultConfigurationCacheIO.kt)
        at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO$readConfigurationCacheStateWithSpecialDecoders$1.invoke(DefaultConfigurationCacheIO.kt)
        at org.gradle.internal.serialize.graph.CodecKt$readWith$1$1.invokeSuspend(Codec.kt:144)
        at org.gradle.internal.serialize.graph.CodecKt$readWith$1$1.invoke(Codec.kt)
        at org.gradle.internal.serialize.graph.CodecKt$readWith$1$1.invoke(Codec.kt)
        at org.gradle.internal.serialize.graph.RunningKt$runReadOperation$2.invokeSuspend(Running.kt:33)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlin.coroutines.ContinuationKt.startCoroutine(Continuation.kt:115)
        at org.gradle.internal.serialize.graph.RunningKt.runToCompletion(Running.kt:54)
        at org.gradle.internal.serialize.graph.RunningKt.runReadOperation(Running.kt:32)
        at org.gradle.internal.serialize.graph.CodecKt.readWith(Codec.kt:143)
        at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO.withReadContextFor(DefaultConfigurationCacheIO.kt:503)
        at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO.withReadContextFor(DefaultConfigurationCacheIO.kt:495)
        at org.gradle.internal.cc.impl.ConfigurationCacheBuildTreeIO$DefaultImpls.withReadContextFor(ConfigurationCacheBuildTreeIO.kt:83)
        at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO.withReadContextFor(DefaultConfigurationCacheIO.kt:94)
        at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO.readConfigurationCacheStateWithSpecialDecoders(DefaultConfigurationCacheIO.kt:349)
        at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO.access$readConfigurationCacheStateWithSpecialDecoders(DefaultConfigurationCacheIO.kt:94)
        at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO$readConfigurationCacheState$1$1.invoke(DefaultConfigurationCacheIO.kt:257)
        at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO$readConfigurationCacheState$1$1.invoke(DefaultConfigurationCacheIO.kt:256)
        at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO.withSharedObjectDecoderFor(DefaultConfigurationCacheIO.kt:333)
        at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO.access$withSharedObjectDecoderFor(DefaultConfigurationCacheIO.kt:94)
        at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO$readConfigurationCacheState$1.invoke(DefaultConfigurationCacheIO.kt:256)
        at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO$readConfigurationCacheState$1.invoke(DefaultConfigurationCacheIO.kt:255)
        at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO.withStringDecoderFor(DefaultConfigurationCacheIO.kt:321)
        at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO.readConfigurationCacheState(DefaultConfigurationCacheIO.kt:255)
        at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO.readRootBuildStateFrom(DefaultConfigurationCacheIO.kt:225)
        at org.gradle.internal.cc.impl.DefaultConfigurationCache$loadWorkGraph$1$1$storeLoadResult$1.invoke(DefaultConfigurationCache.kt:657)
        at org.gradle.internal.cc.impl.DefaultConfigurationCache$loadWorkGraph$1$1$storeLoadResult$1.invoke(DefaultConfigurationCache.kt:656)
        at org.gradle.internal.cc.impl.ConfigurationCacheRepository$StoreImpl$useForStateLoad$1.invoke(ConfigurationCacheRepository.kt:231)
        at org.gradle.internal.cc.impl.ConfigurationCacheRepository$StoreImpl$useForStateLoad$1.invoke(ConfigurationCacheRepository.kt:231)
        at org.gradle.internal.cc.impl.ConfigurationCacheRepository$StoreImpl$useForStateLoad$2.invoke(ConfigurationCacheRepository.kt:239)
        at org.gradle.internal.cc.impl.ConfigurationCacheRepository$StoreImpl$useForStateLoad$2.invoke(ConfigurationCacheRepository.kt:235)
        at org.gradle.internal.cc.impl.ConfigurationCacheRepository$withExclusiveAccessToCache$1.get(ConfigurationCacheRepository.kt:320)
        at org.gradle.cache.internal.LockOnDemandCrossProcessCacheAccess.withFileLock(LockOnDemandCrossProcessCacheAccess.java:90)
        at org.gradle.cache.internal.DefaultCacheCoordinator.withFileLock(DefaultCacheCoordinator.java:213)
        at org.gradle.cache.internal.DefaultPersistentDirectoryStore.withFileLock(DefaultPersistentDirectoryStore.java:147)
        at org.gradle.cache.internal.DefaultCacheFactory$ReferenceTrackingCache.withFileLock(DefaultCacheFactory.java:203)
        at org.gradle.internal.cc.impl.ConfigurationCacheRepository.withExclusiveAccessToCache(ConfigurationCacheRepository.kt:318)
        at org.gradle.internal.cc.impl.ConfigurationCacheRepository.access$withExclusiveAccessToCache(ConfigurationCacheRepository.kt:54)
        at org.gradle.internal.cc.impl.ConfigurationCacheRepository$StoreImpl.useForStateLoad(ConfigurationCacheRepository.kt:235)
        at org.gradle.internal.cc.impl.ConfigurationCacheRepository$StoreImpl.useForStateLoad(ConfigurationCacheRepository.kt:231)
        at org.gradle.internal.cc.impl.DefaultConfigurationCache$loadWorkGraph$1$1.invoke(DefaultConfigurationCache.kt:656)
        at org.gradle.internal.cc.impl.DefaultConfigurationCache$loadWorkGraph$1$1.invoke(DefaultConfigurationCache.kt:655)
        at org.gradle.configurationcache.ConfigurationCacheBuildOperationsKt$withWorkGraphLoadOperation$1.call(ConfigurationCacheBuildOperations.kt:47)
        at org.gradle.configurationcache.ConfigurationCacheBuildOperationsKt$withWorkGraphLoadOperation$1.call(ConfigurationCacheBuildOperations.kt:40)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:210)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:205)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:67)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:60)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:167)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:60)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:54)
        at org.gradle.configurationcache.ConfigurationCacheBuildOperationsKt.withWorkGraphLoadOperation(ConfigurationCacheBuildOperations.kt:40)
        at org.gradle.internal.cc.impl.DefaultConfigurationCache.loadWorkGraph(DefaultConfigurationCache.kt:655)
        at org.gradle.internal.cc.impl.DefaultConfigurationCache.loadRequestedTasks(DefaultConfigurationCache.kt:253)
        at org.gradle.internal.cc.impl.ConfigurationCacheAwareBuildTreeWorkController$scheduleAndRunRequestedTasks$3.apply(ConfigurationCacheAwareBuildTreeWorkController.kt:69)
        at org.gradle.internal.cc.impl.ConfigurationCacheAwareBuildTreeWorkController$scheduleAndRunRequestedTasks$3.apply(ConfigurationCacheAwareBuildTreeWorkController.kt:68)
        at org.gradle.composite.internal.DefaultIncludedBuildTaskGraph.withNewWorkGraph(DefaultIncludedBuildTaskGraph.java:112)
        at org.gradle.internal.cc.impl.ConfigurationCacheAwareBuildTreeWorkController.scheduleAndRunRequestedTasks(ConfigurationCacheAwareBuildTreeWorkController.kt:68)
        at org.gradle.internal.buildtree.DefaultBuildTreeLifecycleController.lambda$scheduleAndRunTasks$1(DefaultBuildTreeLifecycleController.java:77)
        at org.gradle.internal.buildtree.DefaultBuildTreeLifecycleController.lambda$runBuild$4(DefaultBuildTreeLifecycleController.java:120)
        at org.gradle.internal.model.StateTransitionController.lambda$transition$6(StateTransitionController.java:169)
        at org.gradle.internal.model.StateTransitionController.doTransition(StateTransitionController.java:266)
        at org.gradle.internal.model.StateTransitionController.lambda$transition$7(StateTransitionController.java:169)
        at org.gradle.internal.work.DefaultSynchronizer.withLock(DefaultSynchronizer.java:46)
        at org.gradle.internal.model.StateTransitionController.transition(StateTransitionController.java:169)
        at org.gradle.internal.buildtree.DefaultBuildTreeLifecycleController.runBuild(DefaultBuildTreeLifecycleController.java:117)
        at org.gradle.internal.buildtree.DefaultBuildTreeLifecycleController.scheduleAndRunTasks(DefaultBuildTreeLifecycleController.java:77)
        at org.gradle.internal.buildtree.DefaultBuildTreeLifecycleController.scheduleAndRunTasks(DefaultBuildTreeLifecycleController.java:72)
        at org.gradle.tooling.internal.provider.ExecuteBuildActionRunner.run(ExecuteBuildActionRunner.java:31)
        at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
        at org.gradle.internal.buildtree.ProblemReportingBuildActionRunner.run(ProblemReportingBuildActionRunner.java:49)
        at org.gradle.launcher.exec.BuildOutcomeReportingBuildActionRunner.run(BuildOutcomeReportingBuildActionRunner.java:71)
        at org.gradle.tooling.internal.provider.FileSystemWatchingBuildActionRunner.run(FileSystemWatchingBuildActionRunner.java:135)
        at org.gradle.launcher.exec.BuildCompletionNotifyingBuildActionRunner.run(BuildCompletionNotifyingBuildActionRunner.java:41)
        at org.gradle.launcher.exec.RootBuildLifecycleBuildActionExecutor.lambda$execute$0(RootBuildLifecycleBuildActionExecutor.java:54)
        at org.gradle.composite.internal.DefaultRootBuildState.run(DefaultRootBuildState.java:130)
        at org.gradle.launcher.exec.RootBuildLifecycleBuildActionExecutor.execute(RootBuildLifecycleBuildActionExecutor.java:54)
        at org.gradle.internal.buildtree.InitDeprecationLoggingActionExecutor.execute(InitDeprecationLoggingActionExecutor.java:62)
        at org.gradle.internal.buildtree.InitProblems.execute(InitProblems.java:36)
        at org.gradle.internal.buildtree.DefaultBuildTreeContext.execute(DefaultBuildTreeContext.java:40)
        at org.gradle.launcher.exec.BuildTreeLifecycleBuildActionExecutor.lambda$execute$0(BuildTreeLifecycleBuildActionExecutor.java:71)
        at org.gradle.internal.buildtree.BuildTreeState.run(BuildTreeState.java:60)
        at org.gradle.launcher.exec.BuildTreeLifecycleBuildActionExecutor.execute(BuildTreeLifecycleBuildActionExecutor.java:71)
        at org.gradle.launcher.exec.RunAsBuildOperationBuildActionExecutor$2.call(RunAsBuildOperationBuildActionExecutor.java:67)
        at org.gradle.launcher.exec.RunAsBuildOperationBuildActionExecutor$2.call(RunAsBuildOperationBuildActionExecutor.java:63)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:210)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:205)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:67)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:60)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:167)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:60)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:54)
        at org.gradle.launcher.exec.RunAsBuildOperationBuildActionExecutor.execute(RunAsBuildOperationBuildActionExecutor.java:63)
        at org.gradle.launcher.exec.RunAsWorkerThreadBuildActionExecutor.lambda$execute$0(RunAsWorkerThreadBuildActionExecutor.java:36)
        at org.gradle.internal.work.DefaultWorkerLeaseService.withLocks(DefaultWorkerLeaseService.java:263)
        at org.gradle.internal.work.DefaultWorkerLeaseService.runAsWorkerThread(DefaultWorkerLeaseService.java:127)
        at org.gradle.launcher.exec.RunAsWorkerThreadBuildActionExecutor.execute(RunAsWorkerThreadBuildActionExecutor.java:36)
        at org.gradle.tooling.internal.provider.continuous.ContinuousBuildActionExecutor.execute(ContinuousBuildActionExecutor.java:110)
        at org.gradle.tooling.internal.provider.SubscribableBuildActionExecutor.execute(SubscribableBuildActionExecutor.java:64)
        at org.gradle.internal.session.DefaultBuildSessionContext.execute(DefaultBuildSessionContext.java:46)
        at org.gradle.internal.buildprocess.execution.BuildSessionLifecycleBuildActionExecutor$ActionImpl.apply(BuildSessionLifecycleBuildActionExecutor.java:92)
        at org.gradle.internal.buildprocess.execution.BuildSessionLifecycleBuildActionExecutor$ActionImpl.apply(BuildSessionLifecycleBuildActionExecutor.java:80)
        at org.gradle.internal.session.BuildSessionState.run(BuildSessionState.java:73)
        at org.gradle.internal.buildprocess.execution.BuildSessionLifecycleBuildActionExecutor.execute(BuildSessionLifecycleBuildActionExecutor.java:62)
        at org.gradle.internal.buildprocess.execution.BuildSessionLifecycleBuildActionExecutor.execute(BuildSessionLifecycleBuildActionExecutor.java:41)
        at org.gradle.internal.buildprocess.execution.StartParamsValidatingActionExecutor.execute(StartParamsValidatingActionExecutor.java:64)
        at org.gradle.internal.buildprocess.execution.StartParamsValidatingActionExecutor.execute(StartParamsValidatingActionExecutor.java:32)
        at org.gradle.internal.buildprocess.execution.SessionFailureReportingActionExecutor.execute(SessionFailureReportingActionExecutor.java:51)
        at org.gradle.internal.buildprocess.execution.SessionFailureReportingActionExecutor.execute(SessionFailureReportingActionExecutor.java:39)
        at org.gradle.internal.buildprocess.execution.SetupLoggingActionExecutor.execute(SetupLoggingActionExecutor.java:47)
        at org.gradle.internal.buildprocess.execution.SetupLoggingActionExecutor.execute(SetupLoggingActionExecutor.java:31)
        at org.gradle.launcher.daemon.server.exec.ExecuteBuild.doBuild(ExecuteBuild.java:70)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:37)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.WatchForDisconnection.execute(WatchForDisconnection.java:39)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.ResetDeprecationLogger.execute(ResetDeprecationLogger.java:29)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.RequestStopIfSingleUsedDaemon.execute(RequestStopIfSingleUsedDaemon.java:35)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.ForwardClientInput.lambda$execute$0(ForwardClientInput.java:40)
        at org.gradle.internal.daemon.clientinput.ClientInputForwarder.forwardInput(ClientInputForwarder.java:80)
        at org.gradle.launcher.daemon.server.exec.ForwardClientInput.execute(ForwardClientInput.java:37)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.LogAndCheckHealth.execute(LogAndCheckHealth.java:64)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.LogToClient.doBuild(LogToClient.java:63)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:37)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.EstablishBuildEnvironment.doBuild(EstablishBuildEnvironment.java:84)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:37)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
        at org.gradle.launcher.daemon.server.exec.StartBuildOrRespondWithBusy$1.run(StartBuildOrRespondWithBusy.java:52)
        at org.gradle.launcher.daemon.server.DaemonStateCoordinator.lambda$runCommand$0(DaemonStateCoordinator.java:321)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:48)
Caused by: java.io.InvalidObjectException: ReflectiveOperationException during deserialization
        at java.base/jdk.internal.reflect.GeneratedMethodAccessor10.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at org.gradle.internal.serialize.codecs.core.jos.JavaObjectSerializationCodec.readResolve(JavaObjectSerializationCodec.kt:197)
        at org.gradle.internal.serialize.codecs.core.jos.JavaObjectSerializationCodec.decode(JavaObjectSerializationCodec.kt:116)
        at org.gradle.internal.serialize.codecs.core.jos.JavaObjectSerializationCodec$decode$1.invokeSuspend(JavaObjectSerializationCodec.kt)
        ... 229 more
Caused by: java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        ... 234 more
Caused by: java.lang.ClassCastException: class org.gradle.api.internal.file.DefaultCompositeFileTree cannot be cast to class play.twirl.gradle.TwirlSourceDirectorySet (org.gradle.api.internal.file.DefaultCompositeFileTree is in unnamed module of loader org.gradle.internal.classloader.VisitableURLClassLoader @e6ea0c6; play.twirl.gradle.TwirlSourceDirectorySet is in unnamed module of loader org.gradle.internal.classloader.VisitableURLClassLoader$InstrumentingVisitableURLClassLoader @62adf1d8)
        at play.twirl.gradle.TwirlPlugin.$renamedDeserializeLambda$(TwirlPlugin.java:27)
        at play.twirl.gradle.TwirlPlugin.$deserializeLambda$(TwirlPlugin.java)
        ... 237 more
```

--- a/gradle-twirl/build.gradle.kts
+++ b/gradle-twirl/build.gradle.kts
@@ -37,6 +37,7 @@ dependencies {
     compileOnly("org.playframework.twirl:twirl-compiler_2.13:$compilerVersion")
     testImplementation(libs.assertj)
     testImplementation(libs.commons.io)
+    testImplementation(libs.commons.lang)
     testImplementation(libs.freemarker)
 }
 
@@ -59,6 +60,18 @@ testing {
                             val ver = (scalaVersion as String).trimEnd { !it.isDigit() }
                             systemProperty("scala.version", ver)
                         }
+                        // Required to test configuration cache in tests when using withDebug()
+                        // https://github.com/gradle/gradle/issues/22765#issuecomment-1339427241
+                        jvmArgs(
+                            "--add-opens",
+                            "java.base/java.util=ALL-UNNAMED",
+                            "--add-opens",
+                            "java.base/java.util.concurrent.atomic=ALL-UNNAMED",
+                            "--add-opens",
+                            "java.base/java.lang.invoke=ALL-UNNAMED",
+                            "--add-opens",
+                            "java.base/java.net=ALL-UNNAMED",
+                        )
                     }
                 }
             }

--- a/gradle-twirl/gradle/libs.versions.toml
+++ b/gradle-twirl/gradle/libs.versions.toml
@@ -4,11 +4,13 @@ nexus-publish = "2.0.0"
 spotless = "7.0.4"
 assertj = "3.27.3"
 commons-io = "2.19.0"
+commons-lang = "3.17.0"
 freemarker = "2.3.34"
 
 [libraries]
 assertj = { group = "org.assertj", name = "assertj-core", version.ref = "assertj" }
 commons-io = { group = "commons-io", name = "commons-io", version.ref = "commons-io" }
+commons-lang = { group = "org.apache.commons", name = "commons-lang3", version.ref = "commons-lang" }
 freemarker = { group = "org.freemarker", name = "freemarker", version.ref = "freemarker" }
 
 [plugins]

--- a/gradle-twirl/src/main/java/play/twirl/gradle/TwirlPlugin.java
+++ b/gradle-twirl/src/main/java/play/twirl/gradle/TwirlPlugin.java
@@ -10,6 +10,7 @@ import org.gradle.api.Project;
 import org.gradle.api.artifacts.Configuration;
 import org.gradle.api.artifacts.Dependency;
 import org.gradle.api.file.DirectoryProperty;
+import org.gradle.api.file.FileTree;
 import org.gradle.api.internal.lambdas.SerializableLambdas;
 import org.gradle.api.internal.tasks.DefaultSourceSet;
 import org.gradle.api.model.ObjectFactory;
@@ -93,12 +94,14 @@ public class TwirlPlugin implements Plugin<Project> {
               TwirlSourceDirectorySet twirlSource = getTwirlSourceDirectorySet(sourceSet);
               sourceSet.getExtensions().add(TwirlSourceDirectorySet.class, "twirl", twirlSource);
               twirlSource.srcDir(project.file("src/" + sourceSet.getName() + "/twirl"));
+              // See details https://github.com/playframework/twirl/issues/948
+              final FileTree twirlSourceFileTree = twirlSource;
               sourceSet
                   .getResources()
                   .getFilter()
                   .exclude(
                       SerializableLambdas.spec(
-                          (element) -> twirlSource.contains(element.getFile())));
+                          (element) -> twirlSourceFileTree.contains(element.getFile())));
               sourceSet.getAllJava().source(twirlSource);
               sourceSet.getAllSource().source(twirlSource);
 

--- a/gradle-twirl/src/test/java/play/twirl/gradle/AbstractFunctionalTest.java
+++ b/gradle-twirl/src/test/java/play/twirl/gradle/AbstractFunctionalTest.java
@@ -15,6 +15,7 @@ import java.nio.file.Paths;
 import java.util.Map;
 import java.util.stream.Stream;
 import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang3.ArrayUtils;
 import org.gradle.api.JavaVersion;
 import org.gradle.testkit.runner.BuildResult;
 import org.gradle.testkit.runner.GradleRunner;
@@ -95,7 +96,11 @@ abstract class AbstractFunctionalTest {
   }
 
   protected BuildResult build(String gradleVersion, String... args) {
-    return runner.withGradleVersion(gradleVersion).withArguments(args).build();
+    return runner
+        .withGradleVersion(gradleVersion)
+        // Enable configuration cache for any build
+        .withArguments(ArrayUtils.add(args, "--configuration-cache"))
+        .build();
   }
 
   static Stream<String> gradleVersions() {

--- a/gradle-twirl/src/test/java/play/twirl/gradle/SimpleProjectTest.java
+++ b/gradle-twirl/src/test/java/play/twirl/gradle/SimpleProjectTest.java
@@ -142,6 +142,7 @@ public class SimpleProjectTest extends AbstractFunctionalTest {
     Files.copy(projectSourcePath("src/main/twirl/a/b/c.scala.html"), newTemplate);
 
     BuildResult result = build(gradleVersion, "--build-cache", "compileTwirl");
+    assertThat(result.getOutput()).contains("Configuration cache entry stored");
 
     BuildTask compileTwirlResult = result.task(":compileTwirl");
     assertThat(compileTwirlResult).isNotNull();
@@ -160,6 +161,7 @@ public class SimpleProjectTest extends AbstractFunctionalTest {
         .doesNotExist();
 
     result = build(gradleVersion, "--build-cache", "compileTwirl");
+    assertThat(result.getOutput()).contains("Reusing configuration cache");
 
     compileTwirlResult = result.task(":compileTwirl");
     assertThat(compileTwirlResult).isNotNull();

